bench/Hist/Main.hs

```diff
@@ -0,0 +1,472 @@
+      prev <- getParent $ T.pack ver
+
+      runLog <- loadRunLog b exp ver
+      runLogPrev <- loadRunLog b exp $ T.unpack prev
+
+      let diagram = Diagram Live [runLog, runLogPrev] title
+          title = show (unescapeExperiment exp) <> " - live bytes over time compared"
+      plotDiagram True diagram out
+
+  priority 1 $
+    build -/- "*/*.svg" %> \out -> do
+      let [b, ver, exp] = splitDirectories out
+      runLog <- loadRunLog b (Escaped $ dropExtension exp) ver
+      let diagram = Diagram Live [runLog] title
+          title = ver <> " live bytes over time"
+      plotDiagram True diagram out
+
+  build -/- "*.svg" %> \out -> do
+    let exp = Escaped $ dropExtension $ takeFileName out
+    versions <- readVersions
+
+    runLogs <- forM (filter include versions) $ \v -> do
+      loadRunLog build exp $ T.unpack $ humanName v
+
+    let diagram = Diagram Live runLogs title
+        title = show (unescapeExperiment exp) <> " - live bytes over time"
+    plotDiagram False diagram out
+
+----------------------------------------------------------------------------------------------------
+
+data Config = Config
+  { experiments :: [Unescaped String],
+    samples :: Natural,
+    versions :: [GitCommit],
+    -- | Path to the ghcide-bench binary for the experiments
+    ghcideBench :: FilePath,
+    -- | Output folder ('foo' works, 'foo/bar' does not)
+    outputFolder :: String
+  }
+  deriving (Generic, Show)
+  deriving anyclass (FromJSON, ToJSON)
+
+data GitCommit = GitCommit
+  { -- | A git hash, tag or branch name (e.g. v0.1.0)
+    gitName :: Text,
+    -- | A human understandable name (e.g. fix-collisions-leak)
+    name :: Maybe Text,
+    -- | The human understandable name of the parent, if specified explicitly
+    parent :: Maybe Text,
+    -- | Whether to include this version in the top chart
+    include :: Bool
+  }
+  deriving (Binary, Eq, Hashable, Generic, NFData, Show)
+
+instance FromJSON GitCommit where
+  parseJSON (String s) = pure $ GitCommit s Nothing Nothing True
+  parseJSON (Object (toList -> [(name, String gitName)])) =
+    pure $ GitCommit gitName (Just name) Nothing True
+  parseJSON (Object (toList -> [(name, Object props)])) =
+    GitCommit
+      <$> props .:? "git"  .!= name
+      <*> pure (Just name)
+      <*> props .:? "parent"
+      <*> props .:? "include" .!= True
+  parseJSON _ = empty
+
+instance ToJSON GitCommit where
+  toJSON GitCommit {..} =
+    case name of
+      Nothing -> String gitName
+      Just n -> Object $ fromList [(n, String gitName)]
+
+humanName :: GitCommit -> Text
+humanName GitCommit {..} = fromMaybe gitName name
+
+findPrev :: Text -> [GitCommit] -> Text
+findPrev name (x : y : xx)
+  | humanName y == name = humanName x
+  | otherwise = findPrev name (y : xx)
+findPrev name _ = name
+
+----------------------------------------------------------------------------------------------------
+
+-- | A line in the output of -S
+data Frame = Frame
```

Comment:
Same comment as on the other PR. This would probably be cleaner using the machine readable output and now we’re getting to the amount of parsing where it pays off.  Given that this is only used during development however, I don’t mind merging it without that change.

Reply:
The `--machine-readable` option does not work with the detailed GC statistics provided by `-S`, so the only way around here would be event logs

Reply:
Ah that’s a shame :disappointed: